### PR TITLE
[mtouch/tests] Add TimingTests

### PR DESCRIPTION
**Pull request by VincentDondain**
- New MLaunchTool.
- AppLaunchTime (mlaunch): time to launch an application on the simulators.
  How it works: we first open the simulator by launching a dummy app. This allows us to detect if there are any launch watchdogs.
  Therefore, for consistency, all measurements are done with the simulator already open.
  In the case of the AppLaunchTime test, we build the app with the default config and launch it. It's automatically killed by the simulator
  because it does not have a valid entry point but this is fine because it also kills the process and lets us stop the stopwatch.
  We then simply log the time performance.

Note: base commit, baking more tests locally (;

### CODE REVIEW

**Comment by VincentDondain** [comment]
@rolfbjarne I'm very, very open to feedback on this commit <3

File: tests/mtouch/MTouch.cs [hunk 1]
```diff
@@ -627,41 +627,10 @@ public void MT0096 ()
...
-</dict>
-</plist>
-");
-				ExecutionHelper.Execute (TestTarget.ToolPath, string.Format ("-sdkroot " + Configuration.xcode_root + " --sim {0} -sdk {3} --targetver {3} --extension -r:{2} {1}", app, exe, Configuration.XamarinIOSDll, Configuration.sdk_version), hide_output: false);
-				ExecutionHelper.Execute (TestTarget.ToolPath, string.Format ("-sdkroot " + Configuration.xcode_root + " --dev {0} -sdk {3} --targetver {3} --extension -r:{2} {1}", app, exe, Configuration.XamarinIOSDll, Configuration.sdk_version), hide_output: false);
-			} finally {
-				Directory.Delete (testDir, true);
+			using (var mtouch = new MTouchTool ()) {
+				mtouch.CreateTemporaryApp (hasPlist: true);
+				ExecutionHelper.Execute (TestTarget.ToolPath, string.Format ("-sdkroot " + Configuration.xcode_root + " --sim {0} -sdk {3} --targetver {3} --extension -r:{2} {1}", mtouch.AppPath, mtouch.Executable, Configuration.XamarinIOSDll, Configuration.sdk_version), hide_output: false);
```
Comment (rolfbjarne) [inline tests/mtouch/MTouch.cs]: This is half-converted to use `MTouchTool`, could you complete it?

All the corresponding properties should be set, then `Assert.AreEqual (0, mtouch.Execute (sim|dev ...);`

File: tests/mtouch/MTouchTool.cs [hunk 2]
```diff
@@ -326,14 +326,79 @@ public IEnumerable<DeviceInfo> FindAvailableDevices (string [] deviceClasses)
...
+<plist version=""1.0"">
+<dict>
+	<key>CFBundleDisplayName</key>
+	<string>{0}</string>
+	<key>CFBundleIdentifier</key>
+	<string>com.xamarin.{0}</string>
+	<key>CFBundleExecutable</key>
+	<string>{0}</string>
+	<key>MinimumOSVersion</key>
+	<string>6.0</string>
```
Comment (rolfbjarne) [inline tests/mtouch/MTouchTool.cs]: There's a better way than hardcoding versions:  there's a method that will get you the latest version given a profile, just fill in with that (same for the min tvOS version below).

**Comment by rolfbjarne** [comment]
I like it a lot 👍

**Comment by VincentDondain** [comment]
@rolfbjarne `MLaunchTool` isn't complete (watchOS and device support) + `MTouchTool` can be trimmed down but those will be tasks for an other time (;

**Comment by VincentDondain** [comment]
build

**Pull request merged**